mono/metadata/marshal.c

```diff
@@ -7682,6 +7690,83 @@ mono_marshal_get_native_func_wrapper_aot (MonoClass *klass)
 }
 
 /*
+ * mono_marshal_thread_attach:
+ *
+ * Called by native->managed wrappers. Store the original domain which needs to be
+ * restored, or NULL, in dummy.
+ *
+ * Return NULL if it was detached when calling this function, and should return to
+ * blocking mode when detaching
+ */
+static gpointer
+mono_marshal_thread_attach (MonoDomain *domain, gpointer *dummy)
```

Comment:
dummy is the variable name we use in `MONO_PREPARE_RESET_BLOCKING`. And its main purpose is to get the stack address from which to start scanning the native stack.
But I simply reuse it to store the original domain, to avoid having yet another variable.